lib/chef/knife/bmcs_server_create.rb

```diff
@@ -170,6 +177,36 @@ def run
         show_value('Node Name', config[:chef_node_name])
       end
 
+      # Return CreateVnicDetails instance
+      def make_vnic_details
+        vnic_details = OracleBMC::Core::Models::CreateVnicDetails.new
+        vnic_details.assign_public_ip = bool_arg(config[:assign_public_ip]) if config[:assign_public_ip]
+        vnic_details.private_ip = config[:private_ip] if config[:private_ip]
+        vnic_details.subnet_id = config[:subnet_id]
+        vnic_details
+      end
+
+      def get_bootstrap_ip(vnic)
+        # TODO: Consider adding a use_private_ip option.
+        bool_arg(config[:assign_public_ip]) == false ? vnic.private_ip : vnic.public_ip
+      end
+
+      # Return true or false corresponding with boolean string passed in.
+      # Return nil when input is nil.
+      # Error and exit if neither of the above.
+      def bool_arg(arg)
+        case arg
+        when 'yes', 'true'
+          true
+        when 'no', 'false'
+          false
+        when nil
+          nil
+        else
+          error_and_exit('Boolean arguments must be one of: yes, no, true, false')
```

Comment:
I didn't see a boolean argument used this way in the existing codebase.  There are three states I needed to capture:
1) not specified.  uses the default which is decided on the server-side (depending on subnet settings).
2) false - does not assign a public ip.
3) true - does assign a public ip.

I purposely chose a case-sensitive value as otherwise typos are invited as accepted input.  This coincides with (as an example) OpenSSH which is strict about 'yes/no' settings being lowercase.